clients/segments/segments.go

```
@@ -125,44 +119,36 @@ func (c Client) Update(ctx context.Context, id string, body []byte) (api.Respons
 	}
 	err = json.Unmarshal(existing.Data, &getResponse)
 	if err != nil || getResponse.Version == 0 || getResponse.Owner == "" {
-		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+		return api.Response{}, intErr.ErrorValidation{Field: "version, owner", Reason: "invalid"}
```

Here it gets tricky, because we would actually need to transport the information that version OR owner is invalid (as opposed to, maybe, we just fail when BOTH are invalid). Maybe add it to the reason? Instead of "invalid" say "at least one is invalid"?